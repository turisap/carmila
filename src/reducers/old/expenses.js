/**
 * Created by HP on 30-Nov-17.
 */
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
            break;
        case 'REMOVE_EXPENSE':
            return state.filter(({id}) => id !== action.id);
            break;
        case 'EDIT_EXPENSE':
            return state.map(expense => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                }
                return expense;
            });
        default:
            return state;
    }
};