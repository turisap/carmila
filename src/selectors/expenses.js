/**
 * Created by HP on 30-Nov-17.
 */
import moment from 'moment';

const visibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter(exp => {
        const createdAtMoment = moment(exp.createdAt);
        const textMatch = exp.description.toLowerCase().includes(text.toLowerCase());
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch   = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;

        return textMatch && startDateMatch && endDateMatch;
    }).sort((a,b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });

};

export default visibleExpenses;