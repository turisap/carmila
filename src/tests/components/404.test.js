/**
 * Created by HP on 03-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import notFound from '../../components/404';
import expenses from '../fixtures/expenses';

test('should render 404 page',  () => {
    const wrapper = shallow(<notFound />);
    expect(wrapper).toMatchSnapshot();
});