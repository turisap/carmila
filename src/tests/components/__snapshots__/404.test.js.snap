// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 404 page 1`] = `<notFound />`;
